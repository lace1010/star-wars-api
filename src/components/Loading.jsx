import React from "react";
import chewbacca from "../images/chewbacca.png";

const Loading = () => {
  return (
    <div className="fixed top-0 h-screen min-h-[500px] w-full flex items-center justify-center bg-[url('../images/star-wars-bg.jpeg')] mix-blend-soft-light">
      <img
        className="h-52 object-contain  animate-spin "
        src={chewbacca}
        alt="chewbacca head"
      />
    </div>
  );
};

export default Loading;
