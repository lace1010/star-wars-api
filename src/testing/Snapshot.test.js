// import Films from "../pages/Films";
import Loading from "../components/Loading";
import renderer from "react-test-renderer";
// import callSwapi from "../utils/callSwapi";

// this will test loading page because it doesn't wait for data to load
test("app snapshot test", () => {
  const component = renderer.create(<Loading />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// want to run a snapshot test for each page after data has been loaded...
