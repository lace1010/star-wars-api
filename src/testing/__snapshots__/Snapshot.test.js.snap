// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot test 1`] = `
<div
  className="fixed top-0 h-screen min-h-[500px] w-full flex items-center justify-center bg-[url('../images/star-wars-bg.jpeg')] mix-blend-soft-light"
>
  <img
    alt="chewbacca head"
    className="h-52 object-contain  animate-spin "
    src="chewbacca.png"
  />
</div>
`;
